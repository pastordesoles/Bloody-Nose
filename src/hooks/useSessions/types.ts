import { Session } from "../../redux/features/sessionsSlice/types";

interface getAllSessionsResponseBody {
  sessions: {
    isPreviousPage: boolean;
    isNextPage: boolean;
    totalPages: number;
    sessions: Session[];
  };
}

export default getAllSessionsResponseBody;
