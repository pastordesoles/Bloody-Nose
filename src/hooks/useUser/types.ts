import { JwtPayload } from "jwt-decode";

export interface UserCredentials {
  username: string;
  password: string;
}

export interface UserRegisterCredentials extends UserCredentials {
  email: string;
}

export interface LoginResponse {
  token: string;
}

export interface CustomTokenPayload extends JwtPayload {
  username: string;
  id: string;
}
