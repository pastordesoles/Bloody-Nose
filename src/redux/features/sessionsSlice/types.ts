export interface Session {
  picture?: File | string;
  supabasePicture?: string;
  title: string;
  location: string;
  participants: number;
  material: string;
  level: string;
  content: string;
  style: string;
  length: number;
  date: string;
  owner?: string;
  id?: string;
}

export interface SessionsState {
  sessions: Session[];
  session: Session;
}
