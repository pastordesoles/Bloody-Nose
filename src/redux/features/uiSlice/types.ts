export interface UiState {
  showModal: boolean;
  isError: boolean;
  modalText: string;
}

export interface ShowModalActionPayload {
  isError: boolean;
  modalText: string;
}
