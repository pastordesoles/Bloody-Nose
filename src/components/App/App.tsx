import { Routes, Route } from "react-router-dom";

import RegisterPage from "../../pages/RegisterPage/RegisterPage";

function App() {
  return (
    <Routes>
      <Route path="/register" element={<RegisterPage />} />
    </Routes>
  );
}

export default App;
