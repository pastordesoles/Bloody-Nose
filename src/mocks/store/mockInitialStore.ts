import { configureStore } from "@reduxjs/toolkit";
import { userReducer } from "../../redux/features/userSlice/userSlice";
import { store } from "../../redux/store";
import { mockUserStateNotLogged } from "../states/mockUserStates";

const mockInitialStore: typeof store = configureStore({
  reducer: {
    user: userReducer,
  },
  preloadedState: {
    user: mockUserStateNotLogged,
  },
});

export default mockInitialStore;
