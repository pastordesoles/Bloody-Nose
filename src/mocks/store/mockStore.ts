import { configureStore } from "@reduxjs/toolkit";
import { sessionsReducer } from "../../redux/features/sessionsSlice/sessionsSlice";
import { SessionsState } from "../../redux/features/sessionsSlice/types";
import { UiState } from "../../redux/features/uiSlice/types";
import { uiReducer } from "../../redux/features/uiSlice/uiSlice";
import { UserState } from "../../redux/features/userSlice/types";
import { userReducer } from "../../redux/features/userSlice/userSlice";

interface MockStoreProps {
  uiPreloadState?: UiState;
  userPreloadState?: UserState;
  sessionsPreloadState?: SessionsState;
}

const mockStore = ({
  sessionsPreloadState,
  uiPreloadState,
  userPreloadState,
}: MockStoreProps = {}) =>
  configureStore({
    reducer: {
      ui: uiReducer,
      user: userReducer,
      sessions: sessionsReducer,
    },
    preloadedState: {
      ui: uiPreloadState,
      user: userPreloadState,
      sessions: sessionsPreloadState,
    },
  });

export default mockStore;
