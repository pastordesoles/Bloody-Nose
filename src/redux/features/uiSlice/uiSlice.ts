import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { Pagination, ShowModalActionPayload, UiState } from "./types";

const initialState: UiState = {
  isError: false,
  modalText: "",
  showModal: false,
  isLoading: false,
  pagination: {
    currentPage: 0,
    totalPages: 0,
  },
};

const uiSlice = createSlice({
  name: "ui",
  initialState,
  reducers: {
    closeModalReducer: (previousUi: UiState) => ({
      ...previousUi,
      showModal: false,
    }),

    openModalReducer: (
      previousUi: UiState,
      action: PayloadAction<ShowModalActionPayload>
    ) => ({
      ...previousUi,
      showModal: true,
      isError: action.payload.isError,
      modalText: action.payload.modalText,
    }),

    showLoading: (currentUiState) => ({
      ...currentUiState,
      isLoading: true,
    }),

    hideLoading: (currentUiState) => ({
      ...currentUiState,
      isLoading: false,
    }),

    loadPages: (currentUiState, action: PayloadAction<Pagination>) => ({
      ...currentUiState,
      pagination: {
        currentPage: action.payload.currentPage,
        totalPages: action.payload.totalPages,
      },
    }),
    advancePage: (currentUiState) => ({
      ...currentUiState,
      pagination: {
        ...currentUiState.pagination,
        currentPage: currentUiState.pagination.currentPage + 1,
      },
    }),
  },
});

export const uiReducer = uiSlice.reducer;

export const {
  openModalReducer: openModalActionCreator,
  closeModalReducer: closeModalActionCreator,
  showLoading: showLoadingActionCreator,
  hideLoading: hideLoadingActionCreator,
  loadPages: loadPagesActionCreator,
  advancePage: advancePageActionCreator,
} = uiSlice.actions;
