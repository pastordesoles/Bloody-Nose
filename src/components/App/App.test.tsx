import App from "./App";
import { MemoryRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import { Provider } from "react-redux";
import mockInitialStore from "../../mocks/store/mockInitialStore";
import mockStore from "../../mocks/store/mockStore";
import {
  mockUserStateLogged,
  mockUserStateNotLogged,
} from "../../mocks/states/mockUserStates";
import { renderWithProviders } from "../../test-utils/renderWithProviders";
import { screen, waitFor } from "@testing-library/react";
import { UiState } from "../../redux/features/uiSlice/types";

describe("Given an App component", () => {
  describe("When it's rendered with path '/*'", () => {
    test("Then it should render 'Register Page'", () => {
      const expectedApp = TestRenderer.create(
        <Provider store={mockInitialStore}>
          <MemoryRouter initialEntries={["/"]}>
            <App />
          </MemoryRouter>
        </Provider>
      );

      expect(expectedApp).toMatchSnapshot();
    });
  });

  describe("When the user is logged and is currently in the login page", () => {
    test("Then it should be redirected to the sessions page", async () => {
      const store = mockStore({ userPreloadState: mockUserStateLogged });
      const initialEntries = ["/sessions"];
      const title = "Bloody Nose";

      renderWithProviders(<App />, { store, initialEntries });

      await waitFor(() => {
        const heading = screen.queryByRole("heading", {
          name: title,
          level: 1,
        });
        expect(heading).toBeInTheDocument();
      });
    });
  });

  describe("When the user is logged", () => {
    test("Then it should be redirected to the sessions detail page", async () => {
      const store = mockStore({ userPreloadState: mockUserStateLogged });
      const initialEntries = ["/session/:id"];
      const title = "Bloody Nose";

      renderWithProviders(<App />, { store, initialEntries });

      await waitFor(() => {
        const heading = screen.queryByRole("heading", {
          name: title,
          level: 1,
        });
        expect(heading).toBeInTheDocument();
      });
    });
  });

  describe("When the user is logged and goes to an inexistent page", () => {
    test("Then it should be redirected to the not found page", async () => {
      const store = mockStore({ userPreloadState: mockUserStateLogged });
      const initialEntries = ["/ajwirjiew"];
      const title = "Bloody Nose";

      renderWithProviders(<App />, { store, initialEntries });

      await waitFor(() => {
        const heading = screen.queryByRole("heading", {
          name: title,
          level: 1,
        });
        expect(heading).toBeInTheDocument();
      });
    });
  });

  describe("When the user is not logged and is currently in the login page", () => {
    test("Then it should be redirected to the sessions page", () => {
      const store = mockStore({ userPreloadState: mockUserStateNotLogged });
      const initialEntries = ["/sessions"];
      const nameButton = "REGISTER";

      renderWithProviders(<App />, { store, initialEntries });

      const buttonSignUp = screen.getByRole("button", {
        name: nameButton,
      });

      expect(buttonSignUp).toBeInTheDocument();
    });
  });

  describe("When the user is logged and goes to the register page", () => {
    test("Then it should be redirected to the sessions page", () => {
      const store = mockStore({ userPreloadState: mockUserStateLogged });
      const initialEntries = ["/"];
      const title = "Bloody Nose";

      renderWithProviders(<App />, { store, initialEntries });

      const heading = screen.queryByRole("heading", { name: title, level: 1 });

      expect(heading).toBeInTheDocument();
    });
  });

  describe("When there is a success modal", () => {
    test("Then it should show a success modal", () => {
      const mockUiPreloadedState: Partial<UiState> = {
        showModal: true,
        isError: false,
      };

      const store = mockStore({
        uiPreloadState: mockUiPreloadedState as UiState,
      });
      renderWithProviders(<App />, { store });

      const toast = screen.queryByRole("alert");

      expect(toast).toBeInTheDocument();
    });
  });

  describe("When there is an error modal", () => {
    test("Then it should show an error modal", () => {
      const mockUiPreloadedState: Partial<UiState> = {
        showModal: true,
        isError: true,
      };

      const store = mockStore({
        uiPreloadState: mockUiPreloadedState as UiState,
      });
      renderWithProviders(<App />, { store });

      const toast = screen.queryByRole("alert");

      expect(toast).toBeInTheDocument();
    });
  });

  // describe("When the user is logged and goes to the create page", () => {
  //   test("Then it should be redirected to the create page", async () => {
  //     const store = mockStore({ userPreloadState: mockUserStateLogged });
  //     const initialEntries = ["/create"];
  //     const title = "Bloody Nose";

  //     renderWithProviders(<App />, { store, initialEntries });

  //     await waitFor(() => {
  //       const heading = screen.queryByRole("heading", {
  //         name: title,
  //         level: 1,
  //       });
  //       expect(heading).toBeInTheDocument();
  //     });
  //   });
  // });

  describe("When the user is logged and goes to the edit page", () => {
    test("Then it should be redirected to the edit page", async () => {
      const store = mockStore({ userPreloadState: mockUserStateLogged });
      const initialEntries = ["/edit/:id"];
      const title = "Bloody Nose";

      renderWithProviders(<App />, { store, initialEntries });

      await waitFor(() => {
        const heading = screen.queryByRole("heading", {
          name: title,
          level: 1,
        });
        expect(heading).toBeInTheDocument();
      });
    });
  });
});
