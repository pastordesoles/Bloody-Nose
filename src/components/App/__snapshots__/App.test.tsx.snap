// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When it's rendered with path '/*' Then it should render 'Register Page' 1`] = `
<main
  className="sc-dkrFOg jOPgCP"
>
  <div
    className="MuiBox-root css-19r8ggj"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg css-2tlbyn-MuiContainer-root"
    >
      <form
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 form-container css-pyyv1u-MuiPaper-root"
        noValidate={true}
        onSubmit={[Function]}
      >
        <h1
          className="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
        >
          Bloody Nose
        </h1>
        <h2
          className="MuiTypography-root MuiTypography-body1 title__register css-ahj2mt-MuiTypography-root"
        >
          Register
        </h2>
        <div
          aria-label="username"
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root input css-1rmt1wa-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="username"
            id="username-label"
            style={
              Object {
                "color": "#d3d4d9",
              }
            }
          >
            Username
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id="username"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
        <div
          aria-label="password"
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root input css-1rmt1wa-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="password"
            id="password-label"
            style={
              Object {
                "color": "#d3d4d9",
              }
            }
          >
            Password
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id="password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
            />
          </div>
        </div>
        <div
          aria-label="email"
          className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root input css-15zqajq-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
            style={
              Object {
                "color": "#d3d4d9",
              }
            }
          >
            Email
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id="email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth register-button css-1l2mkxo-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "background": "#04395E",
              "fontSize": "1,5rem",
            }
          }
          tabIndex={0}
          type="submit"
        >
          REGISTER
        </button>
        <span
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Already have an account?
           
          <span
            className="login"
          >
            <a
              className="login"
              href="/login"
              onClick={[Function]}
            >
              Login here
            </a>
          </span>
        </span>
      </form>
    </div>
  </div>
</main>
`;
